libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -185,30 +188,36 @@ impl<'a> LiftAstToDml<'a> {
     fn lift_field_type(
         &self,
         ast_field: &ast::Field,
-        custom_type_name: Option<String>,
+        type_alias: Option<String>,
         ast_schema: &ast::SchemaAst,
         checked_types: &mut Vec<String>,
     ) -> Result<(dml::FieldType, Vec<ast::Directive>), DatamodelError> {
         let type_name = &ast_field.field_type.name;
 
         if let Ok(scalar_type) = ScalarType::from_str(type_name) {
             if USE_CONNECTORS_FOR_CUSTOM_TYPES {
-                let pg_connector = BuiltinConnectors::postgres();
-                let args = vec![]; // TODO: figure out args
-                let pg_type_specification = ast_field
-                    .directives
-                    .iter()
-                    .find(|dir| dir.name.name.starts_with("pg.")) // we use find because there should be at max 1.
-                    .map(|dir| dir.name.name.trim_start_matches("pg."));
-
-                if let Some(x) = pg_type_specification.and_then(|ts| pg_connector.calculate_type(&ts, args)) {
-                    let field_type = dml::FieldType::ConnectorSpecific(x);
+                let pg_connector = SqlDatamodelConnectors::postgres();
+                let pg_type_specification = ast_field.directives.iter().find(|dir| dir.name.name.starts_with("pg.")); // we use find because there should be at max 1.
+                let name = pg_type_specification.map(|dir| dir.name.name.trim_start_matches("pg."));
+                let args = pg_type_specification
+                    .map(|dir| {
+                        let args = dir
+                            .arguments
+                            .iter()
+                            .map(|arg| ValueValidator::new(&arg.value).as_int().unwrap() as u32)
+                            .collect();
+                        args
+                    })
+                    .unwrap_or(vec![]);
+
+                if let Some(x) = name.and_then(|ts| pg_connector.parse_native_type(&ts, args)) {
+                    let field_type = dml::FieldType::NativeType(scalar_type, x);
```

Comment:
I know this is a placeholder, but do we already have a way to coerce from a `NativeTypeConstructor`, and do we have access to the connector in this scope?

Reply:
Re 2nd question: Yes the datasource and therefore the connector is in scope.